services/system/Transact/src/RTransact.cpp

```diff
@@ -95,15 +107,34 @@ void RTransact::onBlock()
    auto commitNum  = stat->current.commitNum;
    auto reversible = WriteOnly{}.open<ReversibleBlocksTable>();
    reversible.put({.blockNum = stat->current.blockNum, .time = stat->current.time});
-   BlockTime irreversibleTime = {};
+   BlockTime                      irreversibleTime = {};
+   std::vector<psibase::BlockNum> irreversible;
    for (auto r : reversible.getIndex<0>())
    {
       if (r.blockNum > commitNum)
          break;
       irreversibleTime = r.time;
       if (r.blockNum < commitNum)
+      {
          reversible.remove(r);
+         irreversible.push_back(r.blockNum);
+      }
    }
+
+   // Send replies to all transactions in the block that are now irreversible
+   PSIBASE_SUBJECTIVE_TX
+   {
+      auto blockTracesIdx = Subjective{}.open<BlockTraceTable>().getIndex<0>();
+      for (auto i : irreversible)
```

Comment:
You'll probably need to switch `RTransact` to use the `psibase-service-full-malloc`, instead of `psibase-service-simple-malloc`

Reply:
Why?

Reply:
I was getting `libc++abi: bad_alloc was thrown in -fno-exceptions mode` with full malloc

Reply:
Since multiple traces are being processed in the same execution, the memory usage can get really large if no memory is re-used.